app/models/event.rb

```diff
@@ -224,19 +233,33 @@ def calculate_progress
   end
 
   ##
-  # Returns end of the event
+  # Returns the room in which the event is scheduled
   #
-  def end_time
-    self.start_time + self.event_type.length.minutes
+  def scheduled_room
+    event_schedule = event_schedules.find_by(schedule_id: program.selected_schedule) if program.present? && program.selected_schedule.present?
+    event_schedule.room if event_schedule.present?
+  end
+
+  ##
+  # Returns the start time at which this event is scheduled
+  #
+  def scheduled_start_time
+    event_schedule = event_schedules.find_by(schedule_id: program.selected_schedule) if program.present? && program.selected_schedule.present?
```

Comment:
You do several times
``
event_schedules.find_by(schedule_id: program.selected_schedule)
``
What about introducing a scope for that?